### PR TITLE
feat: kinesis sink

### LINKED ISSUE #10350: support Kinesis sink
### Is your feature request related to a problem? Please describe.

One user is expecting to use Kinesis as they are all in AWS native infrastructure and argues that Kinesis is cheaper than Kafka. 

In terms of AWS's own ecosystem, Kinesis has a lot of native integration with other services, e.g. many DBs, cloud watches, they can all be configured to sink data to Kinesis natively. While if users want to sink to Kafka, they may have to configure a Lambda to do the ingestion if native support is not there. 
We can expect that AWS's customers, a majority, can pick up Kinesis much more easily than Kafka as they almost can never avoid Kinesis.

The user is also willing to use Kinesis sink a lot, which helps us to test the sink.

### Describe the solution you'd like

https://docs.aws.amazon.com/sdk-for-rust/latest/dg/rust_kinesis_code_examples.html

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

closes #10350 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by curiosityyy** [comment]
I am delighted to express my utmost appreciation for the addition of the Kinesis sink feature. This enhancement holds immense significance for my team ([kaito.ai](https://kaito.ai/)), as we heavily rely on Kinesis as our primary data streaming service. The seamless compatibility and exceptional user-friendliness that Kinesis offers in conjunction with other AWS services are truly commendable.

**Comment by lmatz** [comment]
> I am delighted to express my utmost appreciation for the addition of the Kinesis sink feature. This enhancement holds immense significance for my team ([kaito.ai](https://kaito.ai/)), as we heavily rely on Kinesis as our primary data streaming service. The seamless compatibility and exceptional user-friendliness that Kinesis offers in conjunction with other AWS services are truly commendable.

Thank you!
Your appreciation means a lot to us.

File: src/connector/src/sink/kinesis.rs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        Ok(Self {
+            config,
+            schema,
+            pk_indices,
+            client,
+        })
+    }
+
+    pub async fn validate(config: KinesisSinkConfig, pk_indices: Vec<usize>) -> Result<()> {
+        // For upsert Kafka sink, the primary key must be defined.
```
Comment (lmatz) [inline src/connector/src/sink/kinesis.rs:101]: typo: `kafka`

**Review by lmatz — APPROVED** [review]

File: src/connector/src/sink/kinesis.rs (hunk 1 above)
Comment (tabVersion) [reply src/connector/src/sink/kinesis.rs:101]: next time for sure

**Pull request merged**